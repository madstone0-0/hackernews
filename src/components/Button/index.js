/* eslint-disable no-unused-vars */
import React from "react";

function Button(props) {
    const { onClick, className, children } = props;
    return (
        <button type="button" onClick={onClick} className={className}>
            {children}
        </button>

    );
}

Button.defaultProps = {
    className: "",
};

export default Button;