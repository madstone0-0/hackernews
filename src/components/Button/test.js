/* eslint-disable no-unused-vars,no-undef */
/* eslint-disable no-console */
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import React from "react";
import Button from "./index";

describe("Button", () => {
    it("renders", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Button onClick={() => {
            console.log("Clicked");
        }}>Give me more daddy</Button>, div);
    });
});

test("snapshots", () => {
    const component = renderer.create(
        <Button onClick={() => {
            console.log("Clicked");
        }}>Give me more daddy</Button>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});