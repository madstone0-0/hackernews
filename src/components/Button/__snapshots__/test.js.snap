// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Click Me
</button>
`;
